GVFS/GVFS.Common/Maintenance/PackfileMaintenanceStep.cs

```diff
@@ -105,15 +106,20 @@ protected override void PerformMaintenance()
                         activity.RelatedWarning($"Skipping {nameof(PackfileMaintenanceStep)} due to git pids {string.Join(",", processIds)}", Keywords.Telemetry);
                         return;
                     }
-                }
-
-                this.GetPackFilesInfo(out int beforeCount, out long beforeSize, out bool hasKeep);
+                }
+
+                this.GetPackFilesInfo(out int beforeCount, out long beforeSize, out bool hasKeep);
 
                 if (!hasKeep)
                 {
                     activity.RelatedWarning(this.CreateEventMetadata(), "Skipping pack maintenance due to no .keep file.");
                     return;
-                }
+                }
+
+                string multiPackIndexLockPath = Path.Combine(this.Context.Enlistment.GitPackRoot, MultiPackIndexLock);
+                this.Context.FileSystem.TryDeleteFile(multiPackIndexLockPath);

```

Comment:
I see we were doing this before too, but why do we always try to delete the lock file?

Reply:
Git will fail if the .lock file exists, as it thinks another process is writing to that file. It does not inspect handles. 

So, if a power-outage killed our Git process, then that user would never get an updated midx without this deletion of the .lock file.

We use the shared object cache lock to avoid multiple processes modifying this data.